src/Kygekraqmak/KygekPingTPS/PingCommand.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ *     _    __                  _                                     _
+ *    | |  / /                 | |                                   | |
+ *    | | / /                  | |                                   | |
+ *    | |/ / _   _  ____   ____| | ______ ____   _____ ______   ____ | | __
+ *    | |\ \| | | |/ __ \ / __ \ |/ /  __/ __ \ / __  | _  _ \ / __ \| |/ /
+ *    | | \ \ \_| | <__> |  ___/   <| / | <__> | <__| | |\ |\ | <__> |   <
+ * By |_|  \_\__  |\___  |\____|_|\_\_|  \____^_\___  |_||_||_|\____^_\|\_\
+ *              | |    | |                          | |
+ *           ___/ | ___/ |                          | |
+ *          |____/ |____/                           |_|
+ *
+ * A PocketMine-MP plugin to see the server TPS and a player's ping
+ * Copyright (C) 2020-2021 Kygekraqmak
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+*/
+declare(strict_types=1);
+
+namespace Kygekraqmak\KygekPingTPS;
+
+use pocketmine\Player;
+use pocketmine\utils\Config;
+use pocketmine\utils\TextFormat as TF;
+use pocketmine\command\CommandSender;
+use pocketmine\command\Command;
+
+class PingCommand extends Command {
+	
+	/** @var Main $plugin */
+	protected $plugin;
+
+	public function __construct(Main $plugin) {
+		$this->plugin = $plugin;
+		parent::__construct("ping", "Current ping of a player", Main::replace($this->getConfig()->get("prefix") . TF::WHITE . "Usage: /ping <player>"));
+		$this->setPermission("kygekpingtps.ping");
+	}
+
+	private function getConfig() : Config {
+	    return $this->plugin->getConfig();
+    }
+	
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(!$this->testPermission($sender)){
+			$sender->sendMessage(Main::replace($this->getConfig()->get("no-permission", "")));
+			return false;
+		}
+		
+		$target = $this->plugin->getServer()->getPlayer($args[0] ?? "");

```

Comment:
I want the command to show the sender's ping when there's no args.

(You should send command usage instead if the sender is not a player)

Reply:
> I want the command to show the sender's ping when there's no args.
> 
> (You should send command usage instead if the sender is not a player)

well it did give you the sender's ping if you are not the console

Reply:
No it won't.

You did like this
`$target = $this->plugin->getServer()->getPlayer($args[0] ?? "");`
If $args[0] is empty and if a player sends the command, the code will just assume the target player is "", which will _never_ exist.

Please fix the issue.

Reply:
> No it won't.
> 
> You did like this
> `$target = $this->plugin->getServer()->getPlayer($args[0] ?? "");`
> If $args[0] is empty and if a player sends the command, the code will just assume the target player is "", which will _never_ exist.
> 
> Please fix the issue.

`if (is_null($target) && isset($args[0]))`
This will always return true if $args[0] is empty.